src/psyclone/transformations.py

```diff
@@ -477,9 +509,53 @@ def _directive(self, children, collapse=None):
                 "'!$omp taskloop' directives.")
         _directive = OMPTaskloopDirective(children=children,
                                           grainsize=self.omp_grainsize,
-                                          num_tasks=self.omp_num_tasks)
+                                          num_tasks=self.omp_num_tasks,
+                                          nogroup=self.omp_nogroup)
         return _directive
 
+    def apply(self, node, options=None):
+        '''Apply the OMPTaskloopTrans transformation to the specified node in
+        a Schedule. This node must be a Loop since this transformation
+        corresponds to wrapping the generated code with directives like so:
+
+        .. code-block:: fortran
+
+          !$OMP TASKLOOP
+          do ...
+             ...
+          end do
+          !$OMP END TASKLOOP
+
+        At code-generation time (when
+        :py:meth:`OMPLoopDirective.gen_code` is called), this node must be
+        within (i.e. a child of) an OpenMP SERIAL region.
+
+        If the keyword "nogroup" is specified in the options, it will cause a
+        nogroup clause be generated if it is set to True.
+
+        :param node: the supplied node to which we will apply the \
+                     OMPLoopTrans transformation
+        :type node: :py:class:`psyclone.psyir.nodes.Node`
+        :param options: a dictionary with options for transformations\
+                        and validation.
+        :type options: dictionary of string:values or None
+        :param bool options["nogroup"]:
+                indicating whether a nogroup clause should be applied to
+                this taskloop.
+
+        :returns: (:py:class:`psyclone.psyir.nodes.Schedule`, \
+        :py:class:`psyclone.undoredo.Memento`)
+
+        '''
+        if not options:
+            options = {}
+        current_nogroup = self.omp_nogroup
```

Comment:
Please add a comment saying that we allow any nogroup supplied in the options dict to override that supplied to the constructor of the transformation. (Please also mention this in the docstring.)

Reply:
Done